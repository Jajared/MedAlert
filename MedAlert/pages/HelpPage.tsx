import { SafeAreaView, View, Text, StatusBar, StyleSheet, Image } from "react-native";
import BackNavBar from "../components/BackNavBar";

export default function HelpPage({ navigation }) {
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <BackNavBar navigation={navigation} title="Help" />
      <Text style={{ fontSize: 16, alignSelf: "flex-start", marginLeft: 30 }}>Feel free to contact us at:</Text>
      <View style={styles.itemContainer}>
        <Image source={require("../assets/boy.png")} style={styles.icon} />
        <View style={styles.textInfo}>
          <Text style={{ fontWeight: "bold", fontSize: 16 }}>Jared Wong Jing Jie</Text>
          <Text>Y2 Computer Science Student</Text>
          <Text>E0934112@u.nus.edu</Text>
        </View>
      </View>
      <View style={styles.itemContainer}>
        <Image source={require("../assets/girl.png")} style={styles.icon} />
        <View style={styles.textInfo}>
          <Text style={{ fontWeight: "bold", fontSize: 16 }}>Priscilia Pang Qing Wei</Text>
          <Text>Y2 Business Analytics Student</Text>
          <Text>E0968980@u.nus.edu</Text>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#fff",
    flex: 1,
    width: "100%",
  },
  itemContainer: {
    flexDirection: "row",
    backgroundColor: "#FAF0D7",
    borderColor: "#FAF0D7",
    borderWidth: 1,
    borderRadius: 15,
    margin: 15,
    alignItems: "center",
    height: 100,
    width: "90%",
  },
  textContainer: {
    flex: 3,
    flexDirection: "row",
  },
  textInfo: {
    flex: 2,
    paddingHorizontal: 15,
  },
  icon: {
    width: 60,
    height: 60,
    resizeMode: "contain",
    marginLeft: 20,
  },
});
