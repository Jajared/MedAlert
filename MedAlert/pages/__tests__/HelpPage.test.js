import React from "react";
import renderer from "react-test-renderer";

import HelpPage from "../HelpPage";

describe("<HelpPage/>", () => {
  it("has 2 child", () => {
    const tree = renderer.create(<HelpPage />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<HelpPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
