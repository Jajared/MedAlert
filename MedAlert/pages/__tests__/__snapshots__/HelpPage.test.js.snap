// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
  testID="HelpPage"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "width": "100%",
      }
    }
    testID="BackNavBar"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": -20,
          "opacity": 1,
        }
      }
      testID="BackButton"
    >
      <Text />
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 16,
        }
      }
    >
      Help
    </Text>
  </View>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "fontSize": 16,
        "marginLeft": 30,
      }
    }
  >
    Feel free to contact us at:
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FAF0D7",
        "borderColor": "#FAF0D7",
        "borderRadius": 15,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "margin": 15,
        "width": "90%",
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 60,
          "marginLeft": 20,
          "resizeMode": "contain",
          "width": 60,
        }
      }
    />
    <View
      style={
        {
          "flex": 2,
          "paddingHorizontal": 15,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Jared Wong Jing Jie
      </Text>
      <Text>
        Y2 Computer Science Student
      </Text>
      <Text>
        E0934112@u.nus.edu
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FAF0D7",
        "borderColor": "#FAF0D7",
        "borderRadius": 15,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "margin": 15,
        "width": "90%",
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 60,
          "marginLeft": 20,
          "resizeMode": "contain",
          "width": 60,
        }
      }
    />
    <View
      style={
        {
          "flex": 2,
          "paddingHorizontal": 15,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Priscilia Pang Qing Wei
      </Text>
      <Text>
        Y2 Business Analytics Student
      </Text>
      <Text>
        E0968980@u.nus.edu
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
